---
title: C28198 | Документация Майкрософт
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C28198
helpviewer_keywords:
- C28198
ms.assetid: 8bad4acb-712c-43f5-81d1-45d92092d4c5
caps.latest.revision: 5
author: corob-msft
ms.author: corob
manager: jillfra
ms.openlocfilehash: fd469856a808ece4e2906594c8ba3dbe11aef8b5
ms.sourcegitcommit: 68f893f6e472df46f323db34a13a7034dccad25a
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 02/15/2020
ms.locfileid: "77275284"
---
# <a name="c28198"></a>C28198
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

предупреждение C28198: возможна утечка памяти из-за исключения.  
  
 Это предупреждение означает, что выделенная память не освобождается после возникновения исключения. Инструкция в конце пути может вызвать исключение. Память была передана в функцию, которая могла сохранить копию для последующего освобождения.  
  
 Это предупреждение очень похоже на предупреждение [C28197](../code-quality/c28197.md). Здесь также можно использовать заметки, Рекомендуемые для использования с предупреждением [C28197](../code-quality/c28197.md) .  
  
## <a name="example"></a>Пример  
 Следующий пример кода вызывает появление этого предупреждения.  
  
```  
char *p1 = new char[10];  
char *p2 = new char[10];  
  
test(p1);   // does not save a copy of p  
  
delete[] p2;  
delete[] p1;  
```  
  
 Следующий пример кода позволяет избежать появления этого предупреждения.  
  
```  
char *p1 = new char[10];  
char *p2 = NULL;  
  
test(p1);   // does not save a copy of p  
try {  
    p2 = new char[10];  
} catch (std::bad_alloc *e) {  
    // just handle the throw  
    ;  
}  
```
